addons/web_editor/static/src/components/media_dialog/media_dialog.js

```diff
@@ -145,6 +145,7 @@ export class MediaDialog extends Component {
     async save() {
         const selectedMedia = this.selectedMedia[this.state.activeTab];
         if (selectedMedia.length) {
+            //I'm not sure I understand what you are trying to do. Why not do it in the template with a state?
```

Comment:
Hello @FrancoisGe, thank you for going through the code, and making these suggestions :)
I'll take the one from above (in the image_selector.js), 
For this one, there is a better to do things than how they are done for now for sure, the media dialog will take a save callback prop to allow its parent to process the HTMLElement(s) it has generated (based on the user selection). They can be \<img/>, \<span/> for icons, an \<iframe/> for embedded videos, or \<a/> for documents (also, they need to keep some of the attributes of the media that was passed as a prop, if any). 
To do that, each MediaSelector will have its own `createElements` static method to generate the appropriate HTMLElement on save. The comment is about creating these elements, using a qweb template instead of `document.createElement` ?  

For the js/backend/* files comments, I think @Goaman will be able to handle these better than me, as they seem to target code from https://github.com/odoo/odoo/pull/94875